src/dbwrapper.cpp

```diff
@@ -158,14 +158,17 @@ struct CDBBatch::WriteBatchImpl {
 
 CDBBatch::CDBBatch(const CDBWrapper& _parent)
     : parent{_parent},
-      m_impl_batch{std::make_unique<CDBBatch::WriteBatchImpl>()} {};
+      m_impl_batch{std::make_unique<CDBBatch::WriteBatchImpl>()}
+{
+    Clear();
+};
 
 CDBBatch::~CDBBatch() = default;
 
 void CDBBatch::Clear()
 {
     m_impl_batch->batch.Clear();
-    size_estimate = 0;
+    assert(m_impl_batch->batch.ApproximateSize() == kHeader);
```

Comment:
i don't think it's necessary to leave this assertion in the final code.

Reply:
Rebased separately without any change, this was done in https://github.com/bitcoin/bitcoin/compare/4921d9c6254f1a59091120d7720fd86f38ce5bdd..aeda60c600d08e71e51fd251ee0435902caad8c4